### DISCUSSION TITLE
Bridging two servers?

### DISCUSSION THREAD

**Discussion by niclash** [emqx/emqx#7529] [Q&A] [answered]
I have two brokers running, that are unaware of each other.

Does anyone know of any (near) off-the-shelf solution of bridging such brokers, so that everything from a list of topics on broker A is published to same/mapped list of topics on broker B, and vice versa? Doesn't have to be Erlang under the hood.

**Comment by zmstone** [comment, marked as answer]
Hi @niclash 
The EMQX mqtt bridge is essentially an MQTT client implemented inside EMQX.
So it doesn't have to be Erlang under the hood.

If the bridging is bi-directional, you may wan to consider clustering.
If clustering is not an option, you may configure out going (publishing) bridges pointing each other.
doc: https://www.emqx.io/docs/en/v4.4/bridge/bridge-mqtt.html#mqtt-bridge

**Reply by ghost** [reply]
Hi, is there an updated guide for emqx 5?

**Comment by niclash** [comment]
Broker B is not under my control (been around for a long time, no changes possible,++), and IIUIC the MQTT Bridge can't act as a client to Broker B. Right?

**Reply by zmstone** [reply]
Yes. EMQX broker can act like a subscriber client to broker B (the one not under your control).
please refer to the same doc.

**Reply by cnukwas** [reply]
@zmstone we're looking to set up a bi-directional bridge similar to what's discussed here. Are there any changes to this process in 5.x version?

**Reply by keynslug** [reply]
Process is conceptually the same AFAICS, @cnukwas. Please refer to the relevant documentation chapter here: https://www.emqx.io/docs/en/v5.0/data-integration/data-bridge-mqtt.html.
